features/Extended Materials/Shaders/ExtendedMaterials/ExtendedMaterials.hlsli

```diff
@@ -33,28 +33,51 @@ float GetMipLevel(float2 coords, Texture2D<float4> tex)
 }
 
 #if defined(LANDSCAPE)
+#	define HEIGHT_POWER 4.0
+#	define INV_HEIGHT_POWER 0.25
 
-float GetTerrainHeight(PS_INPUT input, float2 coords, float mipLevels[6])
+float GetTerrainHeight(PS_INPUT input, float2 coords, float mipLevels[6], float blendFactor, out float pixelOffset[6])
 {
-	float height = 0.0;
 	if (input.LandBlendWeights1.x > 0.0)
-		height = TexColorSampler.SampleLevel(SampTerrainParallaxSampler, coords, mipLevels[0]).w * input.LandBlendWeights1.x;
+		pixelOffset[0] = pow(input.LandBlendWeights1.x, 1 + 1 * blendFactor) * (pow(TexColorSampler.SampleLevel(SampTerrainParallaxSampler, coords, mipLevels[0]).w, blendFactor * HEIGHT_POWER));
+	else
+		pixelOffset[0] = 0;
```

Comment:
Same above, instead of branching here initialize the whole `pixelOffset` array to zero and save some `else`s.

Reply:
Made it a bit nicer. Idk if there is a faster way to zero-initialize, but I can't leave old values there with inout, otherwise they will be used when weight is 0.